### PR TITLE
fix: replace `InlineEditor` with UIKit input

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-850]

## Test Plan
- Go to setting page, and check if a user can edit username and display name
  - design is slightly changed. User can now confirm/cancel the current change
- Go to experiment detail page, click the menu dropdown, and check if  a user can edit experiment name
  - If a user doesn't have permissions to edit experiment, `edit` button is not visible
  - Check if user cannot edit name, description, and tag by clicking experiment name, description, and tag
- Go to model detail page, click the menu dropdown, and check if  a user can edit model name
  - If a user doesn't have permissions to edit model, `edit` button is disabled
  - Check if user cannot edit name by clicking model name
- Go to model version detail page, click the menu dropdown, and check if  a user can edit model name
  - If a user doesn't have permissions to edit model version, `edit` button is disabled
  - Check if user cannot edit name by clicking model version name

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-850]: https://determinedai.atlassian.net/browse/WEB-850?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/hooks/useModal/Experiment/useModalExperimentEdit.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { ModalFuncProps } from 'antd/es/modal/Modal';
```
Comment (mapmeld) [inline webui/react/src/hooks/useModal/Experiment/useModalExperimentEdit.tsx:1]: why do these files get added in the PR?

Comment (mapmeld) [reply webui/react/src/hooks/useModal/Experiment/useModalExperimentEdit.tsx:1]: oh it's to confirm that you want to change the name of an experiment/model.  will review

**Review by hkang1 — COMMENTED** [review]
I was able to go through the test plan on the preview without any issues. 👍 

I noticed is that the description fields on experiment detail or model view have an different interaction from what you see on Settings Account (username/password fields). Maybe this is more of a design question, but should they be similar?

File: webui/react/src/pages/Settings/SettingsAccount.tsx [hunk 2]
```diff
@@ -78,24 +90,68 @@ const SettingsAccount: React.FC = () => {
...
+              noStyle
+              rules={[{ message: 'Please input your username', required: true }]}>
+              <Input maxLength={32} placeholder="Add username" style={{ widows: '80%' }} />
+            </Form.Item>
+            <Form.Item noStyle>
+              <Button icon={<CloseOutlined />} onClick={() => setIsUsernameEditable(false)} />
+            </Form.Item>
+            <Form.Item noStyle>
+              <Button htmlType="submit" icon={<CheckOutlined />} type="primary" />
+            </Form.Item>
```
[the remark this answers is not in the document]
Comment (keita-determined) [reply webui/react/src/pages/Settings/SettingsAccount.tsx:120]: make sense. i'll update it

**Comment by keita-determined** [comment]
> I was able to go through the test plan on the preview without any issues. 👍
> 
> I noticed is that the description fields on experiment detail or model view have an different interaction from what you see on Settings Account (username/password fields). Maybe this is more of a design question, but should they be similar?

i dont think description fields dont cause any critical issues even if user change it.
However, account setting fields cause login issue like https://determinedai.atlassian.net/browse/WEB-781. that's the reason to add confirmation in the page, but im open to have a discussion about it

**Comment by ClaireNeveu** [comment]
One question: why is the description not part of the "Edit" modal for experiments?

**Comment by hkang1** [comment]
> One question: why is the description not part of the "Edit" modal for experiments?

I think this is a good idea, and the description "field" in the header can just be a display and not interactive.

Separate PR if applicable, but should tags also be handled in the edit modal? Name, description and tags are really the only 3 user editable metadata on the web. Could probably use design input for this.

**Comment by keita-determined** [comment]
> > One question: why is the description not part of the "Edit" modal for experiments?
> 
> I think this is a good idea, and the description "field" in the header can just be a display and not interactive.
> 
> Separate PR if applicable, but should tags also be handled in the edit modal? Name, description and tags are really the only 3 user editable metadata on the web. Could probably use design input for this.

good idea. i can add them in this PR

**Comment by keita-determined** [comment]
Name, description and tags in experiment detail page are in now experiment edit modal.  Name, description and tags in foldable page and title are read only now

**Review by hkang1 — APPROVED** [review]
Looks great Keita!

**Pull request merged**